google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeploy.java

```diff
@@ -37,17 +38,28 @@
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
+import java.io.File;
 import java.lang.reflect.Type;
 import java.nio.file.Path;
 import java.util.Collections;
 import java.util.List;
+import java.util.stream.Collectors;
 
 /**
  * Deploys an application to App Engine.
  */
 public class AppEngineDeploy {
 
   private static final Logger logger = Logger.getInstance(AppEngineDeploy.class);
+  private static final String APPENGINE_GENERATED_DIR = "WEB-INF/appengine-generated/";
+  private static final ImmutableList<String> APPENGINE_CONFIG_FILE_PATHS = ImmutableList.of(
+      APPENGINE_GENERATED_DIR + "index.yaml", // Deploy indexes first since the app needs them.
```

Comment:
since you mentioned index.yaml needs to be deployed first, are you sure the ordering here is honored downstream (e.g. in the core lib etc.)?